OsmAnd/src/net/osmand/plus/server/ApiRouter.java

```diff
@@ -33,21 +30,95 @@ public OsmandApplication getAndroidContext() {
 	private final String FOLDER_NAME = "server";
 	private Gson gson = new Gson();
 	private Map<String, ApiEndpoint> endpoints = new HashMap<>();
+	//change to weakreference
+	public static MapActivity mapActivity;
```

Comment:
Evil, should be variable and should not be weak reference. This class should GC with MapActivity together